tests/keithley_2700.py

```diff
@@ -0,0 +1,188 @@
+        self.ca.set_pv_value("KHLY2700_01:COUNT", 4)
+        self.ca.set_pv_value("KHLY2700_01:LIST_OF_READINGS.PROC", 1)
+        expected_string = self.ca.get_pv_value("KHLY2700_01:LIST_OF_READINGS")
+        self.assertNotEquals(expected_string, "[]")
+
+    def test_GIVEN_sample_count_set_WHEN_read_THEN_sample_count_is_as_expected_AND_within_range(self):
+        expected_sample_count = [2, 0, 5500, 70000]
+        alarm_state = [ChannelAccess.ALARM_NONE, ChannelAccess.ALARM_MAJOR, ChannelAccess.ALARM_NONE,
+                           ChannelAccess.ALARM_MAJOR]
+        for input, output in zip(expected_sample_count, alarm_state):
+            self.ca.set_pv_value("KHLY2700_01:SAMPLE_COUNT:SP", input)
+            self.ca.assert_that_pv_is("KHLY2700_01:SAMPLE_COUNT", input)
+            self.ca.assert_pv_alarm_is("KHLY2700_01:SAMPLE_COUNT", output)
+
+    def test_GIVEN_cycles_rate_set_WHEN_read_THEN_cycle_rate_is_as_expected_AND_within_range(self):
+         expected_cycles = [0.1, 0, 2.0, 65.0]
+         alarm_state = [ChannelAccess.ALARM_NONE, ChannelAccess.ALARM_MAJOR, ChannelAccess.ALARM_NONE,
+                            ChannelAccess.ALARM_MAJOR]
+         for input, output in zip(expected_cycles, alarm_state):
+            self.ca.set_pv_value("KHLY2700_01:NPLCYCLES:SP", input)
+            self.ca.assert_that_pv_is("KHLY2700_01:NPLCYCLES", input)
+            self.ca.assert_pv_alarm_is("KHLY2700_01:NPLCYCLES", output)
+
+    def test_GIVEN_buffer_has_stored_reading_WHEN_queried_next_storable_location_returned(self):
+          expected_location = 11
+          self._lewis.backdoor_set_on_device("buffer", 10)
+          self._ioc.set_simulated_value("KHLY2700_01:SIM:BUFFER_LOC", expected_location)
+
+          self.ca.assert_that_pv_is("KHLY2700_01:BUFFER_LOC", expected_location)
+
+    def test_GIVEN_buffer_state_set_WHEN_read_THEN_buffer_state_is_as_expected(self):
+        sample_data = [0, 1]
+        expected_state = ["ON", "OFF"]
+        for input, output in zip(sample_data, expected_state):
+            self.ca.set_pv_value("KHLY2700_01:BUFFER_STATE:SP", input)
+            self.ca.assert_that_pv_is("KHLY2700_01:BUFFER_STATE", output)
+
+    def test_GIVEN_time_stamp_set_WHEN_read_THEN_time_stamp_is_as_expected(self):
+        sample_data = [0, 1]
+        expected_state = ["ABS", "DELT"]
+        for input, output in zip(sample_data, expected_state):
+            self.ca.set_pv_value("KHLY2700_01:STAMP:SP", input)
+            self.ca.assert_that_pv_is("KHLY2700_01:STAMP", output)
+
+    @skipIf(IOCRegister.uses_rec_sim, "In rec sim this test fails")
+    def test_GIVEN_measurement_resolution_set_WHEN_read_THEN_measurement_resolution_is_as_expected(self):
+        self.ca.set_pv_value("KHLY2700_01:CHANNEL_END:SP", 110)
+        self.ca.set_pv_value("KHLY2700_01:DIGIT_RANGE:SP", 6)
+        self.ca.assert_that_pv_is("KHLY2700_01:DIGIT_RANGE", 6)
+        self.ca.assert_that_pv_is("KHLY2700_01:CHANNEL_END", 110)
```

Comment:
If you use a value more than once, it's best practice to put it in a variable so that if we change it in one place, it changes everywhere